### PR TITLE
DWX-6747 [Hue] Implement Hue SAML LOGOUT URL for CDP

### PR DESCRIPTION
**Pull request by ranade1**
(cherry picked from commit 6a000ed8e83441eb44410e25e6b0f260fbffa174)
(cherry picked from commit e4df9c826a75eac7e8165df7a9411c242324d53e)

## What changes were proposed in this pull request?

- (Please fill in changes proposed in this fix)

## How was this patch tested?

- (Please explain how this patch was tested. Ex: unit tests, manual tests)
- (If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

Please review [Hue Contributing Guide](https://github.com/cloudera/hue/blob/master/CONTRIBUTING.md) before opening a pull request.

### CODE REVIEW

File: desktop/libs/libsaml/src/libsaml/backend.py [hunk 1]
```diff
@@ -120,6 +121,15 @@ def logout(self, request, next_page=None):
       response = saml_logout(request)
       auth_logout(request)
       return response
+    elif conf.LOGOUT_URL.get():
+      auth_logout(request)
+      # This logic was derived from KNOX.
+      prod_url = "console.altus.cloudera.com"
+      redirect_url = "https://sso.cloudera.com/bin/services/support/api/public/logout"
+      if prod_url not in conf.LOGOUT_URL.get():
+        redirect_url = "https://sso.staging-upgrade.aem.cloudera.com/bin/services/support/api/public/logout"
```
Comment (JohanAhlen) [inline desktop/libs/libsaml/src/libsaml/backend.py:130]: The URLs should go in the config

Comment (ranade1) [reply desktop/libs/libsaml/src/libsaml/backend.py:130]: it may change in the future. This will be improved.

**Comment by ranade1** [comment]
recheck

**Comment by ranade1** [comment]
recheck

**Comment by Akhilsnaik** [comment]
> recheck

i guess its due to the python lint error.
Line too long and  Exactly one space required around assignment
```

[03/Nov/2021 09:50:33 -0700] runpylint    INFO     Running pylint with args: /usr/share/hue/build/env/bin/pylint --rcfile=/usr/share/hue/.pylintrc --disable=all --enable=C0326,W0311,C0301 --load-plugins=pylint_django -f parseable desktop/libs/libsaml/src/libsaml/backend.py desktop/libs/libsaml/src/libsaml/conf.py
************* Module libsaml.backend
desktop/libs/libsaml/src/libsaml/backend.py:131: [C0301(line-too-long), ] Line too long (209/140)
desktop/libs/libsaml/src/libsaml/backend.py:131: [C0326(bad-whitespace), ] Exactly one space required around assignment
      html='<html><body onload="document.forms[0].submit()"><form action="%s" method="POST"><input name="logoutRedirect" type="hidden" value="%s"/></form></body></html>' % (conf.LOGOUT_URL.get(), redirect_url)
          ^

-----------------------------------
Your code has been rated at 9.86/10
```

File: desktop/conf.dist/hue.ini [hunk 2]
```diff
@@ -2079,6 +2079,8 @@ submit_to=True
 # Name of the SAML attribute containing the list of groups the user belongs to.
 ## required_groups_attribute=groups
 
+# To log users out of magic-sso, CDP control panel use Logout URL
+## logout_url=
```
Comment (JohanAhlen) [inline desktop/conf.dist/hue.ini:2083]: Perhaps cdp_logout_url as the solution seems very specific to CDP and not SAML?

**Review by JohanAhlen — APPROVED** [review]
Looks good for now. Please do another iteration to extract the URLs, rename the config param and perhaps move it out of the libsaml section as seems very specific to CDP. And some tests would be nice.

**Comment by ranade1** [comment]
recheck

**Comment by JohanAhlen** [comment]
Merged in different PR

**Pull request closed without merging**